design-toolkit/components/src/components/menu/index.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import ChevronRight from '@accelint/icons/chevron-right';
+import { createContext, useCallback, useContext } from 'react';
+import {
+  Header as AriaHeader,
+  Menu as AriaMenu,
+  Collection as AriaMenuCollection,
+  MenuItem as AriaMenuItem,
+  type MenuItemRenderProps as AriaMenuItemRenderProps,
+  MenuSection as AriaMenuSection,
+  MenuTrigger as AriaMenuTrigger,
+  type MenuTriggerProps as AriaMenuTriggerProps,
+  Separator as AriaSeparator,
+  SubmenuTrigger as AriaSubmenuTrigger,
+  Text as AriaText,
+  type ContextValue,
+  Popover,
+} from 'react-aria-components';
+import { callRenderProps, isSlottedContextValue } from '../../lib/utils';
+import { Icon } from '../icon';
+import { MenuStyles, MenuStylesDefaults } from './styles';
+import type {
+  MenuIconProps,
+  MenuItemProps,
+  MenuProps,
+  MenuSectionProps,
+  MenuTextProps,
+  SeparatorProps,
+} from './types';
+
+const { item, label, description, menu, more, icon, separator } = MenuStyles();
+
+export const MenuContext =
+  createContext<ContextValue<MenuProps<unknown>, HTMLDivElement>>(null);
+
+export function Menu<T extends object>(props: MenuProps<T>) {
+  const {
+    children,
+    className,
+    variant = MenuStylesDefaults.variant,
+    selectionMode = 'single',
+    ...rest
+  } = props;
+
+  return (
+    <Popover offset={-4}>
+      <MenuContext.Provider value={{ variant }}>
+        <AriaMenu
+          className={menu({ className })}
+          selectionMode={selectionMode}
+          {...rest}
+        >
+          {children}
+        </AriaMenu>
+      </MenuContext.Provider>
+    </Popover>
+  );
+}
+
+export const MenuTrigger = ({ children }: AriaMenuTriggerProps) => {
+  return <AriaMenuTrigger>{children}</AriaMenuTrigger>;
+};
+MenuTrigger.displayName = 'Menu.Trigger';
+
+export const MenuItem = (props: MenuItemProps) => {
+  const context = useContext(MenuContext);
+  const variant =
+    (isSlottedContextValue(context) ? undefined : context?.variant) ??
+    MenuStylesDefaults.variant;
+
+  const { children: childrenProp, className, ...rest } = props;
+
+  const children = useCallback(
```

Comment:
If we're relying on the React compiler (as I've been told by Jacob), we shouldn't need the useCallback

Reply:
JERIC2O will be a little delayed in adopting the compiler if I had to guess so not a massive harm in including the `useCallback` here. Eventually once all projects are using the compiler we can come back through and remove. We would also need to test the compiler output against this component to see if it is correctly optimizing.